### DISCUSSION TITLE
Cache with paginate

### DISCUSSION THREAD

**Discussion by akulmehta** [Power-Components/livewire-powergrid#547] [Ideas]
It would be nice if there was a caching ability for results. For models with a lot of relationships, the main table and search queries produce a lot of queries and can affect performance/price. Caching could be used in such cases.

The caveat would be how to clear the caches if there is any changes to the data. However, one could create an observer to clear it and make this clear to the developer.

Example https://youtu.be/Ga7-a826i5c

**Comment by luanfreitasdev** [comment]
I support this idea, do you have something in mind?

I believe the cache should be cleared as per the poll too, right?

**Reply by akulmehta** [reply]
> I believe the cache should be cleared as per the poll too, right?

I think the developer can decide that. So it would be like an option in the Table component which they can turn on.



> I support this idea, do you have something in mind?

I may need help writing the code and tests. I will give it a try.
